Makefile

```diff
@@ -30,8 +30,15 @@ protos: compile-protos-go compile-protos-python compile-protos-docs
 
 build: protos build-java build-docker build-html
 
+install-ci-dependencies: install-python-ci-dependencies install-go-ci-dependencies
```

Comment:
Was `install-java-ci-dependencies` needed here too?

Reply:
It's "unimplemented" right now because we actually need some build artifacts like `ingestion` to be available as dependencies. So if we run `install-java-ci-dependencies` it doesn't actually work. Not sure if there is a nice solution that doesn't involve actually building `ingestion` and `datatypes`.

Reply:
https://maven.apache.org/plugins/maven-dependency-plugin/go-offline-mojo.html#excludeGroupIds might be a possible solution.

Reply:
Hmm I see, so you want this image to cache dependencies, ideally.

Good news! After [a decade of the dependency plugin not supporting multi-module builds well for go-offline][1], a swell fellow [fixed it][2] a few months back. The fix is in the latest release, it's working for me after updating, sans `excludeGroupIds`.

I'll try to send a patch when I have a minute to actually test it with the built Docker image, but aside from updating this makefile the patch is just to bump the existing version `3.1.1` in the main `pom.xml` `<pluginManagement>` section to `3.1.2`.

Might need to remove the `cd`s into the subdirectories, I don't think Maven deals with it well—just `mvn dependency:go-offline` from top level should do it.

[1]: https://issues.apache.org/jira/browse/MDEP-204
[2]: https://github.com/apache/maven-dependency-plugin/pull/20

Reply:
Above doesn't work, also tried https://github.com/qaware/go-offline-maven-plugin

I think I will just use `verify` for now